src/crimson/common/fixed_kv_node_layout.h

```diff
@@ -558,6 +558,19 @@ class FixedKVNodeLayout {
     set_meta(Meta::merge_from(left.get_meta(), right.get_meta()));
   }
 
+  static uint32_t get_balance_pivot_idx(
+    const FixedKVNodeLayout &left,
+    const FixedKVNodeLayout &right,
+    bool prefer_left)
```

Comment:
I see in the fixed-btree, `prefer_left==false` means the removed item was in the right, else it was in the left.

But I can't recall why `pivot_idx` needs to be processed like this, do you remember why
```c++
if (total % 2 && prefer_left) {
  pivot_idx++;
}
```
?

Reply:
It seems that `prefer_left` indicates whether the donor of the balancing is from the left size. If it is, `prefer_left` is false; otherwise, it's true.

Reply:
@athanatos  do you remember why this (`prefer_left` and `pivot_idx++`) is needed foundamentally?

Reply:
The intent seems to that we keep the odd element on the node to which we are trying to balance elements (the one which just had a removal).  So, if the donor is on the left, we set prefer_left=false to put the extra on the right.  I can't actually remember whether this is strictly necessary, or simply that we expect more removals on the balance target.

Reply:
I can't think of a reason why it would be required now -- as long as the donor has more than min putting the extra on either should still satisfy the invariant that both have at least min.  I might simply have misunderstood an edge case at the time.

Reply:
I've just went through `Introduction to algorithms, 3rd edition, Thomas H. Cormen`.

According to chapter 18.3, when deleting a key from a node, if the node has only `t - 1` keys and one of its siblings has `t` keys, the node should borrow one key from that sibling. So, following this, I think the `prefer_left` has its value in making sure to borrow one key from the right sibling if it has exacly `t` keys.

However, this does seem unnecessary as it just change the case from `t-1-key left, t-key right` to `t-key left, t-1-key right`. But I'm not sure if this is really unnecessary, it's textbook after all. So I'm throw this here.

Reply:
On the other hand, it seems that the current lba tree may inevitablly hit the case of removing a key from a minimal-occupancy node, since it handles the merge after removing the key. This may be violating the b+ tree invariant?

So I think maybe we should change it to handle the merge before removing the key. Does this make sense to you guys? @cyx1231st @athanatos